setup.py

```diff
@@ -239,6 +239,7 @@ def run(self):
 extras_require['all'] = everything
 
 if 'setuptools' in sys.modules:
+    setuptools_extra_args['python_requires'] = '>=3.3'
```

Comment:
Don't you need to require this on TravisCI? https://github.com/ipython/ipython/blob/master/.travis.yml#L13

Btw, should not be something to detect/request this requirement at the setup.py level?